### PR TITLE
make scheme configurable, include full service DNS name

### PR DESCRIPTION
**Pull request by melindaloubser1**
- make scheme (http/s) configurable for rasa, rasa-x and duckling
-  include full service DNS name in host paths 

Problem trying to solve: Providing https urls for model server/config endpoint etc. 
Why we can't just set the env var in values.yml: the default certs are valid only for the service DNS name. The full service DNS name needs {{ .Release.Namespace }} to be resolved, which isn't accessible from values.yml.

### CODE REVIEW

**Comment by melindaloubser1** [comment]
@erohmensing fyi
@tczekajlo Definitely not convinced this is how I'm supposed to do this.

File: charts/rasa-x/values.yaml [hunk 1]
```diff
@@ -114,6 +116,8 @@ rasa:
   tag: ""
   # port on which Rasa runs
   port: 5005
+  # scheme
+  scheme: https
```
Comment (tczekajlo) [inline charts/rasa-x/values.yaml:120]: `https`?

File: charts/rasa-x/templates/rasa-x-config-files-configmap.yaml [hunk 2]
```diff
@@ -8,8 +8,8 @@ data:
   environments: |
     rasa:
       production:
-        url: "http://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaProduction.serviceName }}:{{ .Values.rasa.port }}"
+        url: "{{ default "HTTP" .Values.rasa.scheme }}://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaProduction.serviceName }}:{{ .Values.rasa.port }}"
```
Comment (tczekajlo) [inline charts/rasa-x/templates/rasa-x-config-files-configmap.yaml:11]: Why here don't you use `my-svc.my-namespace.svc` name?

File: charts/rasa-x/values.yaml (hunk 1 above)
Comment (melindaloubser1) [reply charts/rasa-x/values.yaml:120]: oops!

File: charts/rasa-x/templates/rasa-x-config-files-configmap.yaml (hunk 2 above)
Comment (melindaloubser1) [reply charts/rasa-x/templates/rasa-x-config-files-configmap.yaml:11]: my bad!

File: charts/rasa-x/templates/rasa-config-files-configmap.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ metadata:
 data:
   rasa-credentials: |
     rasa:
-      url: http://{{ include "rasa-x.host" . }}:{{ .Values.rasax.port }}/api
+      url: {{ default "HTTP" .Values.rasax.scheme }}://{{ include "rasa-x.host" . }}.{{ .Release.Namespace }}.svc:{{ .Values.rasax.port }}/api
```
Comment (tczekajlo) [inline charts/rasa-x/templates/rasa-config-files-configmap.yaml:10]: We don't need `default` here. The `scheme` parameter has a value in the `values.yaml` file.

**Review by tczekajlo — APPROVED** [review]
I've left one commend 😄

File: charts/rasa-x/templates/rasa-config-files-configmap.yaml (hunk 3 above)
Comment (melindaloubser1) [reply charts/rasa-x/templates/rasa-config-files-configmap.yaml:10]:
```suggestion
      url: {{  .Values.rasax.scheme }}://{{ include "rasa-x.host" . }}.{{ .Release.Namespace }}.svc:{{ .Values.rasax.port }}/api
```

File: charts/rasa-x/templates/rasa-config-files-configmap.yaml [hunk 4]
```diff
@@ -45,7 +45,7 @@ data:
       {{- end }}
       {{- end }}
     action_endpoint:
-      url: "http://{{ include "rasa-x.fullname" . }}-app:{{ template "rasa-x.custom-actions.port" . }}{{ .Values.app.endpoints.actionEndpointUrl }}"
+      url: "{{ default "HTTP" .Values.app.scheme }}://{{ include "rasa-x.fullname" . }}-app.{{ .Release.Namespace }}.svc:{{ template "rasa-x.custom-actions.port" . }}{{ .Values.app.endpoints.actionEndpointUrl }}"
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-config-files-configmap.yaml:48]:
```suggestion
      url: "{{  .Values.app.scheme }}://{{ include "rasa-x.fullname" . }}-app.{{ .Release.Namespace }}.svc:{{ template "rasa-x.custom-actions.port" . }}{{ .Values.app.endpoints.actionEndpointUrl }}"
```

File: charts/rasa-x/templates/rasa-deployments.yaml [hunk 5]
```diff
@@ -67,7 +67,7 @@ spec:
         - --no-prompt
         - --production
         - --config-endpoint
-        - "http://{{ include "rasa-x.fullname" $ }}-rasa-x:{{ default 5002 $.Values.rasax.port }}/api/config?token=$(RASA_X_TOKEN)"
+        - {{ default "HTTP" $.Values.rasax.scheme }}://{{ include "rasa-x.fullname" $ }}-rasa-x.{{ $.Release.Namespace }}.svc:{{ default 5002 $.Values.rasax.port }}/api/config?token=$(RASA_X_TOKEN)
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-deployments.yaml:70]:
```suggestion
        - {{  $.Values.rasax.scheme }}://{{ include "rasa-x.fullname" $ }}-rasa-x.{{ $.Release.Namespace }}.svc:{{ default 5002 $.Values.rasax.port }}/api/config?token=$(RASA_X_TOKEN)
```

File: charts/rasa-x/templates/rasa-deployments.yaml [hunk 6]
```diff
@@ -135,9 +135,9 @@ spec:
         - name: "RASA_ENVIRONMENT"
           value: "{{ .rasaEnvironment }}"
         - name: "RASA_MODEL_SERVER"
-          value: "http://{{ include "rasa-x.host" $ }}:{{ $.Values.rasax.port }}/api/projects/default/models/tags/{{ .modelTag }}"
+          value: {{ default "HTTP" $.Values.rasax.scheme }}://{{ include "rasa-x.host" $ }}.{{ $.Release.Namespace }}.svc:{{ $.Values.rasax.port }}/api/projects/default/models/tags/{{ .modelTag }}
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-deployments.yaml:138]:
```suggestion
          value: {{ $.Values.rasax.scheme }}://{{ include "rasa-x.host" $ }}.{{ $.Release.Namespace }}.svc:{{ $.Values.rasax.port }}/api/projects/default/models/tags/{{ .modelTag }}
```

File: charts/rasa-x/templates/rasa-deployments.yaml [hunk 7]
```diff
@@ -135,9 +135,9 @@ spec:
...
         - name: "RASA_DUCKLING_HTTP_URL"
-          value: "http://{{ include "duckling.host" $ }}:{{ $.Values.duckling.port }}"
+          value: {{ default "HTTP" $.Values.duckling.scheme }}://{{ include "duckling.host" $ }}.{{ $.Release.Namespace }}.svc:{{ $.Values.duckling.port }}
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-deployments.yaml:140]:
```suggestion
          value: {{ $.Values.duckling.scheme }}://{{ include "duckling.host" $ }}.{{ $.Release.Namespace }}.svc:{{ $.Values.duckling.port }}
```

File: charts/rasa-x/templates/rasa-x-config-files-configmap.yaml [hunk 8]
```diff
@@ -8,8 +8,8 @@ data:
   environments: |
     rasa:
       production:
-        url: "http://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaProduction.serviceName }}:{{ .Values.rasa.port }}"
+        url: "{{ default "HTTP" .Values.rasa.scheme }}://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaProduction.serviceName }}.{{ .Release.Namespace }}.svc:{{ .Values.rasa.port }}"
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-x-config-files-configmap.yaml:11]:
```suggestion
        url: "{{ .Values.rasa.scheme }}://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaProduction.serviceName }}.{{ .Release.Namespace }}.svc:{{ .Values.rasa.port }}"
```

File: charts/rasa-x/templates/rasa-x-config-files-configmap.yaml [hunk 9]
```diff
@@ -8,8 +8,8 @@ data:
...
         token: ${RASA_TOKEN}
       worker:
-        url: "http://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaWorker.serviceName }}:{{ .Values.rasa.port }}"
+        url: "{{ default "HTTP" .Values.rasa.scheme }}://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaWorker.serviceName }}:{{ .Values.rasa.port }}"
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-x-config-files-configmap.yaml:14]:
```suggestion
        url: "{{ .Values.rasa.scheme }}://{{ include "rasa-x.fullname" . }}-{{ .Values.rasa.versions.rasaWorker.serviceName }}:{{ .Values.rasa.port }}"
```

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 10]
```diff
@@ -82,7 +82,7 @@ spec:
         - name: "LOCAL_MODE" # This variable doesn't do anything anymore in Rasa X 0.28 and later
           value: "false"
         - name: "RASA_X_HOST"
-          value: "http://{{ include "rasa-x.host" . }}:{{ .Values.rasax.port }}"
+          value: {{ default "HTTP" .Values.rasax.scheme }}://{{ include "rasa-x.host" . }}.{{ .Release.Namespace }}.svc:{{ .Values.rasax.port }}
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-x-deployment.yaml:85]:
```suggestion
          value: {{ .Values.rasax.scheme }}://{{ include "rasa-x.host" . }}.{{ .Release.Namespace }}.svc:{{ .Values.rasax.port }}
```

**Comment by tczekajlo** [comment]
@melindaloubser1 is it ready to merge? 😄

**Comment by erohmensing** [comment]
@tczekajlo does it then make sense to take the probe scheme from this scheme? 😄 or just keep it that the user needs to define it

File: charts/rasa-x/templates/rasa-config-files-configmap.yaml [hunk 11]
```diff
@@ -45,7 +45,7 @@ data:
       {{- end }}
       {{- end }}
     action_endpoint:
-      url: "http://{{ include "rasa-x.fullname" . }}-app:{{ template "rasa-x.custom-actions.port" . }}{{ .Values.app.endpoints.actionEndpointUrl }}"
+      url: "{{  .Values.app.scheme }}://{{ include "rasa-x.fullname" . }}-app.{{ .Release.Namespace }}.svc:{{ template "rasa-x.custom-actions.port" . }}{{ .Values.app.endpoints.actionEndpointUrl }}"
```
Comment (erohmensing) [inline charts/rasa-x/templates/rasa-config-files-configmap.yaml:48]: we should also add this to the values.yaml for app, right?

Comment (tczekajlo) [reply charts/rasa-x/templates/rasa-config-files-configmap.yaml:48]: Yes! Good 👀  @erohmensing 😄

**Review by tczekajlo — CHANGES_REQUESTED** [review]

**Review by tczekajlo — APPROVED** [review]
LGTM 🚀

**Pull request merged**